src/state/apis/swapper/helpers/validateTradeQuote.ts

```diff
@@ -221,9 +225,24 @@ export const validateTradeQuote = async (
     // Swappers other than THORChain shouldn't be affected by this limitation
     if (swapperName !== SwapperName.Thorchain) return false
 
-    // Sender is either a smart contract address, or the bytecode is still loading - disable confirm
-    const _isSmartContractAddress = await isSmartContractAddress(sendAddress)
-    if (_isSmartContractAddress !== false) return true
+    // This is either a smart contract address, or the bytecode is still loading - disable confirm
+    const _isSmartContractSellAddress = await isSmartContractAddress(sendAddress)
+    const _isSmartContractReceiveAddress = await isSmartContractAddress(quote.receiveAddress)
+    // For long-tails, the *destination* address cannot be a smart contract
+    // https://dev.thorchain.org/aggregators/aggregator-overview.html#admonition-warning
+    // This doesn't apply to regular THOR swaps however, which docs have no mention of *destination* having to be an EOA
+    // https://dev.thorchain.org/protocol-development/chain-clients/evm-chains.html?search=smart%20contract
+    if (
+      [firstHop.source, secondHop.source].some(source =>
+        [THORCHAIN_LONGTAIL_SWAP_SOURCE, THORCHAIN_LONGTAIL_STREAMING_SWAP_SOURCE].includes(source),
+      ) &&
+      _isSmartContractReceiveAddress !== false
+    )
```

Comment:
I think we've now broken quotes from THORChain. The network response returns correctly, but we get the below (with and without a custom receive address):

<img width="1015" alt="Screenshot 2024-02-05 at 9 26 30 am" src="https://github.com/shapeshift/web/assets/97164662/3a19b3b0-a852-4008-90e3-aae242fa2b1f">

Reply:
That one was actually quite the rabbit-hole! tl;dr here is we had absolutely no type-safety on `steps.length` which could (according to types) be of any n-length. Fixed this by bringing in the context of single and multi-hop trade quotes, the former consisting of 1 step, the latter consisting of 2 (for now, as we only have a max. of two hops here).
This now brings type errors which should've been present before e.g in here - 1bbd41ad7f46bd18fd0b8946595788113feb19b4